slave.mk

```diff
@@ -648,7 +648,7 @@ $(addprefix $(FILES_PATH)/, $(SONIC_MAKE_FILES)) : $(FILES_PATH)/% : .platform $
 		# Remove target to force rebuild
 		rm -f $(addprefix $(FILES_PATH)/, $*)
 		# Apply series of patches if exist
-		if [ -f $($*_SRC_PATH).patch/series ]; then pushd $($*_SRC_PATH) && quilt applied || QUILT_PATCHES=../$(notdir $($*_SRC_PATH)).patch quilt push -a; popd; fi $(LOG)
+		if [ -f $($*_SRC_PATH).patch/series ]; then pushd $($*_SRC_PATH) && QUILT_PATCHES=../$(notdir $($*_SRC_PATH)).patch quilt push -a || quilt push -a 2>&1 | grep -q "series fully applied"; popd; fi $(LOG)
```

Comment:
>quilt push -a [](http://example.com/codeflow?start=135&length=13)

Why retry?
Why removing the env var?
Is it simply reverting [https://github.com/sonic-net/sonic-buildimage/pull/13386](https://github.com/sonic-net/sonic-buildimage/pull/13386) ?

Reply:
Alternatively, we may check if any patch already applied, if yes, revert back, and apply all again.

Reply:
The second one only checks whether the patch applied. It is to keep the output information (STDOUT/ERROUT) the same with the original one.
The second one does not need the env QUILT_PATCHES, because the .pc file has already been created by the first command, so the variable is not required.

Reply:
> Alternatively, we may check if any patch already applied, if yes, revert back, and apply all again.

Do you think it is good below? Looks like to remove and apply again is not necesary, right?
```
if [ -f $($*_SRC_PATH).patch/series ]; then pushd $($*_SRC_PATH) && { quilt pop -a 1>/dev/null 2>&1 || ture } && QUILT_PATCHES=../$(notdir $($*_SRC_PATH)).patch quilt push -a; popd; fi $(LOG)
```

Reply:
I prefer no retry. You last full command has a `pushd` which is not needed.

Reply:
Qi, thanks for your comment, will change it to remove all patches before applying patches.